server/TypeDBServer.java

```diff
@@ -109,14 +111,28 @@ protected TypeDBServer(CoreConfig config, boolean debug, Factory factory) {
         server = rpcServer();
         Thread.setDefaultUncaughtExceptionHandler(
                 (t, e) -> {
-                    logger().error(UNCAUGHT_EXCEPTION.message(t.getName(), e), e);
-                    close();
-                    System.exit(1);
+                    try {
+                        logger().error(UNCAUGHT_ERROR.message(t.getName(), e), e);
+                        close();
+                        System.exit(1);
+                    } catch (TypeDBCheckedException ex) {
+                        logger().error("Failed to shut down cleanly, performing hard stop.");
+                        Runtime.getRuntime().halt(1);
```

Comment:
if this block triggers, the thread that is shutting down failed to shut down cleanly (because of OOM or other reasons) so we force a halt